src/khoj/processor/conversation/openai/utils.py

```diff
@@ -100,7 +100,10 @@ def completion_with_backoff(
     # Calculate cost of chat
     input_tokens = chunk.usage.prompt_tokens if hasattr(chunk, "usage") and chunk.usage else 0
     output_tokens = chunk.usage.completion_tokens if hasattr(chunk, "usage") and chunk.usage else 0
-    cost = chunk.usage.model_extra.get("estimated_cost") or 0  # Estimated costs returned by DeepInfra API
+    if chunk.usage:
```

Comment:
Wouldn't we want to make sure that `chunk.usage.model_extra` is present, not just `chunk.usage`? Otherwise, you would have the same problem again as the code in the above lines assumes `chunk.usage`. For the error in the stack trace you've shared, the issue seems to specifically be due to the `model_extra` field.

```suggestion
    if chunk.usage.model_extra
```

Same for the change below.

Reply:
the newest pull has fix this you can merge that .